sda/local.go

```diff
@@ -243,60 +242,48 @@ func (l *LocalTest) GetServices(hosts []*Host, sid ServiceID) []Service {
 	return services
 }
 
-// MakeHELS is an abbreviation to make a Host, an Roster, and a service.
-// It returns the service of the first host in the slice.
+// MakeHELS creates nbr hosts, and will return the associated roster. It also
+// returns the Service object of the first hosts in the list having sid as a
+// ServiceID.
 func (l *LocalTest) MakeHELS(nbr int, sid ServiceID) ([]*Host, *Roster, Service) {
-	hosts := l.GenLocalHosts(nbr, false, true)
+	hosts := l.GenHosts(nbr, false, true)
 	el := l.GenRosterFromHost(hosts...)
 	return hosts, el, l.Services[hosts[0].ServerIdentity.ID][sid]
 }
 
-// NewLocalHost creates a new host with the given address and registers it.
-func NewLocalHost(port int) *Host {
+// NewPrivIdentity returns a secret + ServerIdentity. The SI will have
+// "localhost:+port as first address.
+func NewPrivIdentity(port int) (abstract.Scalar, *network.ServerIdentity) {
 	address := "localhost:" + strconv.Itoa(port)
 	priv, pub := PrivPub()
 	id := network.NewServerIdentity(pub, address)
+	return priv, id
+}
+
+// NewTCPHost creates a new host with a tcpRouter with "localhost:"+port as an
+// address.
+func NewTCPHost(port int) *Host {
+	priv, id := NewPrivIdentity(port)
 	return NewHost(id, priv)
 }
 
-// GenLocalHosts will create n hosts with the first one being connected to each of
-// the other nodes if connect is true.
-func GenLocalHosts(n int, connect bool, processMessages bool) []*Host {
+// NewLocalHost returns a new host using a LocalRouter (channels) to communicate.
+// At the return of this function, the router is already Run()ing in a go
+// routine.
```

Comment:
Yep. It just made me realizes that I can even remove all of these suspicious magical boolean flags on theses methods (`connect, process, register bool`...)